### DISCUSSION TITLE
My tokio won't send data on Ubuntu

### DISCUSSION THREAD

**Discussion by fu050409** [tokio-rs/tokio#6426] [Q&A]
I have a strange problem, I made a custom TCP based end-to-end encryption protocol, however when using `stream.write_all` and  `stream.write` on Ubuntu, the client didn't receive any data, but the connection was normal and the server was able to receive data normally.

My code passed the test locally on my computer (I have Windows) and I tested it locally on my Ubuntu host without any anomalies. I tried running my code on a Windows remote host and I was equally successful (remote access and local access).

For what it's worth, I also wrote a Python implementation for it, and it works fine on my Ubuntu server. It interfaces correctly with the Rust client.

I'm not sure why this is, I'm using the latest version of Rust and tokio at the moment.

Forgive me if I asked a stupid question.

**Comment by Darksonn** [comment]
You have not really provided very much information. Perhaps you need to flush after writing?

**Reply by fu050409** [reply]
Yes, I tried `stream.flush`, but it still not work.

**Reply by fu050409** [reply]
I tried this approach and even if I do a `flush` after every `write` method call it doesn't work:

```
match stream.write_all(&data).await {
        Ok(_) => {
            match stream.flush().await {
                Ok(_) => {}
                Err(e) => {
                    return Err(OblivionException::UnsupportedMethod {
                        error: e.to_string(),
                    });
                }
            };
        }
        Err(e) => {
            return Err(OblivionException::UnsupportedMethod {
                error: e.to_string(),
        });
    }
};
```

**Reply by fu050409** [reply]
My Ubuntu is  `Ubuntu 22.04.3 LTS (GNU/Linux 5.15.0-84-generic x86_64)`

**Reply by fu050409** [reply]
If you need any more information feel free to let me know, I've been confused by this for a long time

**Reply by fu050409** [reply]
I've been trying for a long time and I think it might be a server operator issue. I apologize for the disturbance.
